core/src/main/scala/org/http4s/multipart/Multipart.scala

```diff
@@ -19,7 +19,16 @@ package multipart
 
 import org.http4s.headers._
 
-final case class Multipart[F[_]](parts: Vector[Part[F]], boundary: Boundary = Boundary.create) {
+/** Create a new multipart from a vector of parts and a boundary.
+  *
+  * Caution: the default argument for boundary is impure.  There is no
+  * binary-compatible fix, but the default parameter will be dropped
+  * in 1.0.
+  */
+final case class Multipart[F[_]](
+    parts: Vector[Part[F]],
+    boundary: Boundary = Boundary.unsafeCreate(),
```

Comment:
I can't figure out a binary compatible way to deprecate just the default parameter.  The signature is otherwise what we want.

Reply:
Ooh, I can give this a try.

Reply:
I tried adding a `apply$default$2()` or whatever it is, and was still in synthetic hell.  Might be able to figure it out with javap, but I gave up.

Reply:
Btw, if we take away the default parameter it will definitely be source-breaking. Is that ok?

A source-preserving strategy would be to deprecate this `apply` and add a new one with the arguments in reverse order.

Reply:
That's an intriguing idea, but inconsistent with the ordering of the fields of the case class.  I'm not sure that's a good deal.